.github/workflows/ci.yml

```diff
@@ -1,16 +1,81 @@
 name: CI
+
 on:
   push:
-    branches: [main, release]
+    branches: [main, release/*]
   pull_request:
-concurrency:
-  group: ${{ github.workflow }}-${{ github.ref }}
-  cancel-in-progress: true
+    branches: [main]
+
 jobs:
-  ci:
-    name: CI
+  changed-files:
     runs-on: ubuntu-latest
-    if: github.event_name != 'push' || !startsWith(github.event.head_commit.message, 'v')
+    outputs:
+      files: ${{ steps.changed-files.outputs.all_modified_files	}}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Detect changed files
+        uses: tj-actions/changed-files@v35
+        id: changed-files-dir-names-max-depth
+        with:
+          dir_names: "true"
+          dir_names_max_depth: 1
+
+  ci-web:
+    needs: changed-files
+    if: contains(needs.changed-files.outputs.files, 'web/')
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: web
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          node-version: lts/*
+      - uses: actions/setup-node@v3
+        with:
+          node-version-file: '.nvmrc'
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+      - uses: actions/cache@v3
+        id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+      - name: Install
+        run: yarn install
+      - name: Check
+        run: yarn run check
+      - name: Test
+        run: yarn run coverage
+      - name: Check translations
+        run: yarn i18n --fail-on-update
+      - name: Send coverage report
+        uses: codecov/codecov-action@v2
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          flags: web
+      - name: Build
+        run: yarn build
+      - name: Pack
+        if: github.ref == 'refs/heads/main' || github.ref == 'refs/heads/release'
+        run: mv dist reearth-web && tar -zcvf reearth-web.tar.gz reearth-web
+      - uses: actions/upload-artifact@v2
+        if: github.ref == 'refs/heads/main' || github.ref == 'refs/heads/release'
+        with:
+          name: build
+          path: web/reearth-web.tar.gz
+          if-no-files-found: error
```

Comment:
Added from [reearth-web](https://github.com/reearth/reearth-web/blob/main/.github/workflows/ci.yml) with a little modifications in the end including making a build, checking coverage etc.